cola/widgets/dag.py

```diff
@@ -191,6 +209,11 @@ def context_menu_event(self, event):
         menu.addSeparator()
         menu.addAction(self.menu_actions['create_branch'])
         menu.addAction(self.menu_actions['create_tag'])
+        reset_menu = menu.addMenu(N_('Reset'))
+        reset_menu.setEnabled(self.menu_actions['reset_soft'].isEnabled())
```

Comment:
The menu disabling is a nice touch here. Thanks for the careful attention to usability details.